File: solutions/remove_element_27/rust_solution/remove_element.rs
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut slowIdx = 0;
        for pos in (0..nums.len()) {
            if nums[pos] != val {
                nums[slowIdx] = nums[pos];
                slowIdx += 1;
            }
        }
        return slowIdx as i32;
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_remove_element() {
        let mut nums1 = vec![3,2,2,3];
        assert_eq!(Solution::remove_element(&mut nums1, 3), 2);

        let mut nums2 = vec![0,1,2,2,3,0,4,2];
        assert_eq!(Solution::remove_element(&mut nums2, 2), 5);
    }
}

File: solutions/Array/RemoveElement_27/remove_element.py
from typing import List


class Solution:
    def remove_element(self, nums: List[int], val: int) -> int:
        """
        >>> Solution().remove_element([3,2,2,3], 3)
        2
        >>> Solution().remove_element([0,1,2,2,3,0,4,2], 2)
        5
        """
        fast, slow = 0, 0
        while fast < len(nums):
            if nums[fast] != val:
                nums[slow] = nums[fast]
                slow += 1
            fast += 1
        return slow
        
